libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -817,6 +817,14 @@ bool GCS_MAVLINK::handle_mission_item(mavlink_message_t *msg, AP_Mission &missio
         result = MAV_MISSION_INVALID_SEQUENCE;
         goto mission_ack;
     }
+
+    // sanity check for DO_JUMP command
+    if (cmd.id == MAV_CMD_DO_JUMP) {
+        if (cmd.content.jump.target >= waypoint_request_last || cmd.content.jump.target == 0) {
```

Comment:
I don't think this is valid as it doesn't take into account partial mission uploads. IE I can have 100 waypoints uploaded, and want to swap waypoint 40-50 out with 41 being a jump to 90. waypoint_request_last will be set to 50 in this case and we would reject a valid command. This check should instead be based around checking if the command is >= waypoint_request_last or if its >= the mission total number of commands.